apps/console/src/features/roles/components/wizard-updated/role-basics.tsx

```diff
@@ -0,0 +1,388 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+
+import { Alert, ListItemText } from "@oxygen-ui/react";
+import { IdentifiableComponentInterface } from "@wso2is/core/models";
+import { Field, Form } from "@wso2is/form";
+import { Link } from "@wso2is/react-components";
+import { AxiosResponse } from "axios";
+import debounce, { DebouncedFunc } from "lodash-es/debounce";
+import isEmpty from "lodash-es/isEmpty";
+import React, {
+    FunctionComponent,
+    MutableRefObject,
+    ReactElement,
+    SyntheticEvent,
+    useCallback,
+    useEffect,
+    useRef,
+    useState
+} from "react";
+import { Trans, useTranslation } from "react-i18next";
+import { DropdownProps } from "semantic-ui-react";
+import { useApplicationList } from "../../../applications/api/application";
+import { ApplicationListItemInterface } from "../../../applications/models";
+import { history } from "../../../core";
+import { AppConstants } from "../../../core/constants";
+import { searchRoleList } from "../../api/roles";
+import { RoleConstants } from "../../constants";
+import { CreateRoleFormData, RoleAudiences, SearchRoleInterface } from "../../models";
+
+const FORM_ID: string = "add-role-basics-form";
+
+/**
+ * Interface to capture role basics props.
+ */
+interface RoleBasicProps extends IdentifiableComponentInterface {
+    dummyProp?: string;
+    /**
+     * Initial values of the form.
+     */
+    initialValues: CreateRoleFormData;
+    /**
+     * Trigger submission
+     */
+    triggerSubmission: (submitFunctionCb: () => void) => void;
+    /**
+     * On submit callback.
+     */
+    onSubmit: (values: CreateRoleFormData) => void;
+    /**
+     * Set whether the next button is disabled.
+     */
+    setIsNextDisabled: (isDisabled: boolean) => void;
+}
+
+/**
+ * Component to capture basic details of a new role.
+ *
+ * @param props - Role Basic prop types
+ */
+export const RoleBasics: FunctionComponent<RoleBasicProps> = (props: RoleBasicProps): ReactElement => {
+
+    const {
+        onSubmit,
+        triggerSubmission,
+        initialValues,
+        setIsNextDisabled,
+        [ "data-componentid" ]: componentId
+    } = props;
+
+    const { t } = useTranslation();
+
+    const [ roleAudience, setRoleAudience ] = useState<string>(RoleConstants.DEFAULT_ROLE_AUDIENCE);
+    const [ isDisplayApplicationList, setIsDisplayApplicationList ] = useState<boolean>(false);
+    const [ isDisplayNoAppScopeApplicatioError, setIsDisplayNoAppScopeApplicatioError ] = useState<boolean>(false);
+    const [ isFormError, setIsFormError ] = useState<boolean>(false);
+    const [ applicationSearchQuery, setApplicationSearchQuery ] = useState<string>(undefined);
```

Comment:
This should be `undefined` as this will be passed to the `useApplicationList`, if it is an empty string it will pass the filter even we do not want it.